lib/apiservers/engine/proxy/archive.go

```diff
@@ -182,6 +187,8 @@ func (a *ArchiveProxy) ArchiveImportWriter(ctx context.Context, store, deviceID
 		} else {
 			pipeReader.Close()
 		}
+		op.Debugf("Stream for device %s has returned from PL. Err received is %v ", deviceID, plErr)
+		errchan <- plErr
```

Comment:
given this _must_ happen for the `Close` call to exit it's good practice to put it in a defer for safety - ensures someone doesn't accidentally add in an exit path that skips it.